### PR TITLE
Prepare for v0.16 and v1.0

### PR DESCRIPTION
**Pull request by meyskens**
/hold

This prepares the CI for the next release :tada:

### CODE REVIEW

File: config/jobs/cert-manager/cert-manager-presubmits.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ presubmits:
     decorate: true
     branches:
     - master
-    - release-0.16
+    - release-1.0
```
Comment (wallrj) [inline config/jobs/cert-manager/cert-manager-presubmits.yaml:63]: Does this configure prow to run this test before merging into either of these two branches?
Well then what about if we want to do a 0.16 patch release? It looks like we're configuring prow to no longer run these tests before merging stuff into the release-0.16 branch. Or have I got the wrong end of the stick?

File: config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ presubmits:
     agent: kubernetes
     decorate: true
     branches:
-    - release-0.15
+    - release-0.16
```
Comment (wallrj) [inline config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml:12]: Ah, I guess this answers my earlier question.

File: config/jobs/cert-manager/release-previous/cert-manager-release-previous-periodics.yaml [hunk 3]
```diff
@@ -479,3 +479,61 @@ periodics:
       options:
       - name: ndots
         value: "1"
+
+- name: ci-cert-manager-previous-e2e-v1-18
```
Comment (wallrj) [inline config/jobs/cert-manager/release-previous/cert-manager-release-previous-periodics.yaml:483]: Does this mean we hadn't been periodically running e2e tests for CM 0.15 on k8s 1.18?
Was that just an oversight? Or is there a reason that 0.15 wasn't tested on k8s 1.18?

Comment (meyskens) [reply config/jobs/cert-manager/release-previous/cert-manager-release-previous-periodics.yaml:483]: we did not as on time of release k8s was not released, we did always test master with it

**Comment by meyskens** [comment]
@wallrj for now it is edited by hand, a script would be nice but not always is the change this small

**Pull request merged**